lib/navigation/navigator.hpp

```diff
@@ -1,34 +1,19 @@
-#pragma once
+#include "consts.hpp"
 
-#include <core/types.hpp>
+namespace hyped::navigation{
 
-namespace hyped::navigation {
+class Navigator{
 
-struct Trajectory {
-  core::Float displacement;
-  core::Float velocity;
-  core::Float acceleration;
-};
+  public:
+
+  Navigator();
+
+  void navigate();
+
+  Trajectory trajectory_;
```

Comment:
This shouldn't be public?

Reply:
Wasn't sure what format data would be published by the navigation system over ROS so left it public until it was decided. Do you think nav should publish trajectory elements separately instead?

Reply:
This is not a question about ROS. If you want to expose it, do it in a function. Otherwise there is external write access to this class member!

Reply:
Fair point...